xwiki-pro-macros-test/xwiki-pro-macros-test-docker/src/test/it/com/xwiki/pro/test/ui/GenericMacrosIT.java

```diff
@@ -76,61 +101,582 @@
+        assertEquals("solid " + hexToRgb("#f536f5"), panel2.getBorderStyle());
+        assertNull(panel2.getWidth());
+        assertNull(panel2.getHeight());
+        assertNull(panel2.getBorderRadius());
+
+        // Title section 2nd panel.
+        assertEquals("NestedPanelsTestTitle", panel2.getTitleText());
+        assertNull(panel2.getTitleBackgroundColor());
+        assertNull(panel2.getTitleColor());
+
+        // Content section 2nd panel.
+        assertEquals("Content3 for PanelMacroTest\nNestedPanelsTestTitle2\nContent4 for PanelMacroTest",
+            panel2.getContentText());
+        assertNull(panel2.getContentBackgroundColor());
+        assertNull(panel2.getContentColor());
+
+        // Checks the CSS class of the 3rd panel.
+        assertTrue(panel3.getCssClass().contains("testCssClass"));
+
+        // Panel container 3rd panel.
+        assertEquals("groove rgb(153, 0, 153)", panel3.getBorderStyle());
+        assertNull(panel3.getWidth());
+        assertNull(panel3.getHeight());
+        assertNull(panel3.getBorderRadius());
+
+        // Title section 3rd panel.
+        assertEquals("NestedPanelsTestTitle2", panel3.getTitleText());
+        assertNull(panel3.getTitleBackgroundColor());
+        assertNull(panel3.getTitleColor());
+
+        // Content section 3rd panel.
+        assertEquals("Content4 for PanelMacroTest", panel3.getContentText());
+        assertNull(panel3.getContentBackgroundColor());
+        assertNull(panel3.getContentColor());
+    }
+
+    @Test
+    @Order(8)
+    void microsoftStreamMacroTest(TestUtils setup, TestReference testReference)
+    {
+        setup.createPage(testReference, createContent("microsoftstream-macros.vm"), "MicrosoftStreamTest");
+
+        MicrosoftStreamMacroPage page = new MicrosoftStreamMacroPage();
+
+        MicrosoftStreamMacro mStream1 = page.getMacro(0);
+        MicrosoftStreamMacro mStream2 = page.getMacro(1);
+        MicrosoftStreamMacro mStream3 = page.getMacro(2);
+
+        // There should be 3 MicrosoftStream macros.
+        assertEquals(3, page.getMacroCount());
+
+        // Checks the alignment of the macro.
+        assertEquals("right", mStream1.getAlignment());
+
+        // Checks the width of the macro, default = "500px".
+        assertEquals("600px", mStream1.getWidth());
+
+        // Checks the height of the macro, default = "300px".
+        assertEquals("400px", mStream1.getHeight());
+
+        // Checks the autoplay property, default = "false".
+        assertTrue(mStream1.hasAutoplay());
+        // Checks the showInfo property, default = "false".
+        assertTrue(mStream1.hasShowInfo());
+        // Checks the StartTime property existence.
+        assertTrue(mStream1.hasStartTime());
+        // Checks the StartTime property.
+        assertEquals(convertStartTime("01:12:13"), mStream1.getStartTime());
+        // Checks the actual URL of the microsoftStream.
+        assertTrue(mStream1.hasCorrectURL("www.youtube.com"));
```

Comment:
This is a minor thing, but microsoftStreamMacro is not for youtube links. I know this is just an example, but since you added something real, you should use something correct like `https://web.microsoftstream.com` (to be checked). Or just add something dummy `www.stream.com`